src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef, useState } from "react";
+import ModalBasic from "./ModalBasic/ModalBasic";
+import styles from "./Modal.module.scss";
+import { useOnClickOutside } from "usehooks-ts";
+
+interface Modal {
+  name: string;
+  subtitle?: string;
+  shareButtons?: boolean;
+}
+
+function Modal({ name, ...rest }: Modal) {
+  // 평상시 modal 노출 false
+  const [isOpen, setIsOpen] = useState(false);
+  const modalRef = useRef<HTMLDivElement>(null);
+  // 모달 닫기 false
+  const handleClickOutside = () => {
```

Comment:
해당 함수가 어떤 동작을 하는지 표현해주는게 좋을 것 같아요~
함수 내부 동작으로만 보면 toggle 느낌이긴 한데, 실제로 이 함수를 사용하는 쪽을 보면
모달을 닫는 용도로 사용하는 것 같아서 `handleClickCloseModal`같은 느낌도 괜찮을 것 같네요~
```suggestion
  const handleClickToggleModal = () => {
```